frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -597,6 +602,37 @@ export default function SurveyEdit(): JSX.Element {
                                 </LemonField.Pure>
                             ),
                         },
+                        {
+                            key: SurveyEditSection.CompletionConditions,
+                            header: 'Completion conditions',
+                            content: (
+                                <div className="flex flex-row gap-2 items-center">
+                                    <LemonCheckbox
+                                        checked={!!survey.responses_limit}
+                                        onChange={(checked) => {
+                                            const newResponsesLimit = checked ? 100 : null
+                                            setSurveyValue('responses_limit', newResponsesLimit)
+                                        }}
+                                    />
+                                    Stop the survey once received
+                                    <LemonInput
+                                        type="number"
+                                        size="small"
+                                        min={1}
+                                        value={survey.responses_limit || NaN}
+                                        onChange={(value) => {
+                                            if (value && value > 0) {
+                                                setSurveyValue('responses_limit', value)
+                                            } else {
```

Comment:
Using `setSurveyValue` should be unnecessary here, follow the managed component, like here:

https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/surveys/SurveyEdit.tsx#L359 (i.e. use value and onChange)